tests/test_cli_compile.py

```diff
@@ -427,11 +451,21 @@ def test_editable_package_without_non_editable_duplicate(pip_conf, runner):
     assert "small-fake-a==" not in out.stderr
 
 
-def test_editable_package_constraint_without_non_editable_duplicate(pip_conf, runner):
+def test_editable_package_constraint_without_non_editable_duplicate(
+    pip_conf, runner, current_resolver
+):
     """
     piptools keeps editable constraint,
     without also adding a duplicate "non-editable" requirement variation
     """
+
+    if current_resolver != RESOLVER_LEGACY:
+        pytest.skip(
+            "This test is actual only for legacy resolver. "
+            "Constraints refactored in 2020 resolver. "
+            "See https://github.com/pypa/pip/issues/9020 for details."
+        )
+
```

Comment:
It's way cleaner like this:
```suggestion
@pytest.mark.parametrize(
    'current_resolver',
    (
       RESOLVER_LEGACY,
    ),
    indirect=('current_resolver', ),
)
@pytest.mark.usefixtures('current_resolver')
def test_editable_package_constraint_without_non_editable_duplicate(pip_conf, runner):
    """
    piptools keeps editable constraint,
    without also adding a duplicate "non-editable" requirement variation
    """
```
and won't produce unnecessary skipped statuses in the pytest report.

Reply:
You could also make it reusable like this:
```suggestion

legacy_resolver_only = pytest.mark.parametrize(
    'current_resolver',
    (
       RESOLVER_LEGACY,
    ),
    indirect=('current_resolver', ),
)

@legacy_resolver_only
@pytest.mark.usefixtures('current_resolver')
def test_editable_package_constraint_without_non_editable_duplicate(pip_conf, runner):
    """
    piptools keeps editable constraint,
    without also adding a duplicate "non-editable" requirement variation
    """
```

Reply:
Addressed in c6f0efbfff7ab513c6e6c2e25e90700527f797d8